### PR TITLE
Added PickupSystem and integrated ECS parent-child

### PR DESCRIPTION
**Pull request by Peter-JanGootzen**
# Description

During the creation of [the magnificent parent-child PR](https://github.com/Brick-Studios/BrickEngine/pull/40) we needed a proper way to functionally test the incredibly high quality code that had been written, it was of such sophistication that we needed an actual ECS system that would use the new functionality. Which is why we created a PickupSystem which is theory should be able to be used with items, weapons and whatever weapons of mass destruction one might want to use.
This PR also integrates the changes that were made to the engine in [the elusive parent-child PR](https://github.com/Brick-Studios/BrickEngine/pull/40).

## Screenshot(s)
_If you can't see it, there is a weapon near the Gorilla._
2019-11-10-112131_1600x900_scrot
2019-11-10-112134_1600x900_scrot

# How can this be tested?

You can press E to pickup or drop a weapon with the Gorilla. Test it with [the incomprehensibly amazing parent-child PR](https://github.com/Brick-Studios/BrickEngine/pull/40)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Included 1 or more screenshots
- [x] Code is const correct

### CODE REVIEW

File: include/entities/entity_factory.hpp [hunk 1]
```diff
@@ -14,10 +14,11 @@ class EntityFactory {
 public:
     EntityFactory(std::shared_ptr<EntityManager> em, RenderableFactory& rf);
     ~EntityFactory() = default;
-    int createPanda(double x, double y, int playerId) const;
-    int createGorilla(double x, double y, int playerId) const;
-    int createImage(std::string path, int x, int y, int width, int height, Layers layer, int alpha);
-    int createPlatform(double x, double y, double xScale, double yScale, std::string path, int alpha);
+    int createPanda(double xPos, double yPos, double xScale, double yScale, int playerId) const;
+    int createGorilla(double xPos, double yPos, double xScale, double yScale, int playerId) const;
+    int createWeapon(double xPos, double yPos, double xScale, double yScale) const;
```
Comment (LuukSantegoeds) [inline include/entities/entity_factory.hpp]: You probably do not want scale here

File: lib/components/pickup_component.cpp [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#include "components/pickup_component.hpp"
+
+PickupComponent::PickupComponent(){}
+
+std::string PickupComponent::getNameStatic(){
+    return "pickupcomponent";
```
Comment (LuukSantegoeds) [inline lib/components/pickup_component.cpp]: Change to "PickupComponent"

File: include/entities/entity_factory.hpp (hunk 1 above)
Comment (LuukSantegoeds) [inline include/entities/entity_factory.hpp]: createWeapon needs to be able to support other weapons in the future. Change method to createPistol or make use of a parameter

**Review by LuukSantegoeds — CHANGES_REQUESTED** [review]

File: include/entities/entity_factory.hpp (hunk 1 above)
Comment (JanSchollaert) [reply include/entities/entity_factory.hpp]: PJ & me were discussing about refactoring the whole entity_factory, a lot of things could be improved but that might be better in a different pull request

Comment (Peter-JanGootzen) [reply include/entities/entity_factory.hpp]: I don't think it should be decided how to create the different weapons right now, this is just a test weapon for the PickupSystem. When the actual weapons will get created and loaded from the json, the _right_ way on how to create those will become apparent.

As for reworking the EntityFactory, the functions at are final like panda and gorilla are good, I think we should fix this factory up whenever we need it.

**Review by LuukSantegoeds — APPROVED** [review]
[**APPROVED**](https://media.giphy.com/media/YKdm91XONLTaw/giphy.gif)

File: include/systems/pickup_system.hpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+#include "systems/beast_system.hpp"
+#include "brickengine/collision_detector.hpp"
+
+class PickupSystem : public BeastSystem {
+public:
+    PickupSystem(std::shared_ptr<CollisionDetector> cd, std::shared_ptr<EntityManager> em, std::shared_ptr<EntityFactory> ef);
+    void update(double delta_time);
+    ~PickupSystem() = default;
+private:
+    std::shared_ptr<CollisionDetector> collisionDetector;
```
Comment (Bram-Boris) [inline include/systems/pickup_system.hpp]: snek

File: lib/systems/pickup_system.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
+#include "brickengine/input.hpp"
+#include "brickengine/components/player_component.hpp"
+
+#include "systems/pickup_system.hpp"
+#include "components/pickup_component.hpp"
+#include "player_input.hpp"
+
+PickupSystem::PickupSystem(std::shared_ptr<CollisionDetector> cd,
+    std::shared_ptr<EntityManager> entityManager, std::shared_ptr<EntityFactory> ef)
```
Comment (Bram-Boris) [inline lib/systems/pickup_system.cpp]: snek

**Review by Bram-Boris — COMMENTED** [review]
Just some sneks :+1:

**Review by Bram-Boris — APPROVED** [review]

**Pull request merged**